File: src/glureau/kdp/creational/abstractfactory/java/ConcreteFactoryB.java
package glureau.kdp.creational.abstractfactory.java;

/**
 * Concrete implementation for AbstractFactory class: this class implements the
 * creational methods for all the products of a specific family
 */
class ConcreteFactoryB implements AbstractFactory {

	@Override
	public AbstractProduct createProduct() {
		System.out.println("Called: implementation of factory B; will return: ConcreteProductB instance");
		return new ConcreteProductB();
	}
}

File: src/glureau/kdp/creational/abstractfactory/kotlin/ConcreteFactoryB.kt
package glureau.kdp.creational.abstractfactory.kotlin

/**
 * Concrete implementation for AbstractFactory class: this class implements the
 * creational methods for all the products of a specific family
 */
internal class ConcreteFactoryB : AbstractFactory {

    override fun createProduct(): AbstractProduct {
        println("Called: implementation of factory B; will return: ConcreteProductB instance")
        return ConcreteProductB()
    }
}
